File: 1.Fundamental-Level/1.Advanced-C#/02.Multidimensional-Arrays-Sets-Dictionaries/07.Phonebook/Phonebook.cs
namespace _07.Phonebook
{
    using System;
    using System.Collections.Generic;

    public class Phonebook
    {
        public static void Main()
        {
            var phonebook = new Dictionary<string, HashSet<string>>();

            FillPhonebook(phonebook);

            while (true)
            {
                string searchName = Console.ReadLine();

                if (string.IsNullOrWhiteSpace(searchName))
                {
                    break;
                }

                if (phonebook.ContainsKey(searchName))
                {
                    Console.WriteLine("{0} -> {1}", searchName, string.Join(", ", phonebook[searchName]));
                }
                else
                {
                    Console.WriteLine($"Contact {searchName} does not exist.");
                }
            }
        }

        private static void FillPhonebook(IDictionary<string, HashSet<string>> phonebook)
        {
            string line = Console.ReadLine();
            while (line != "search")
            {
                string[] lineParams = line.Split('-');
                string name = lineParams[0];
                string number = lineParams[1];

                if (!phonebook.ContainsKey(name))
                {
                    phonebook[name] = new HashSet<string>();
                }

                phonebook[name].Add(number);

                line = Console.ReadLine();
            }
        }
    }
}

File: Java-Development/Advanced-Java/02.Sets-and-Maps-Exercises/src/_05_Phonebook/Phonebook.java
package _05_Phonebook;

import java.util.HashMap;
import java.util.Scanner;

public class Phonebook {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String line = scanner.nextLine();

        HashMap<String, String> phoneBook = new HashMap<>();
        while (!"search".equals(line)) {
            String[] lineArgs = line.split("-");
            String name = lineArgs[0];
            String phone = lineArgs[1];
            phoneBook.put(name, phone);

            line = scanner.nextLine();
        }

        line = scanner.nextLine();
        while (!"stop".equals(line)) {
            String nameToSearch = line;
            if (phoneBook.containsKey(nameToSearch)) {
                System.out.printf("%s -> %s%n", nameToSearch, phoneBook.get(nameToSearch));
            } else {
                System.out.printf("Contact %s does not exist.%n", nameToSearch);
            }

            line = scanner.nextLine();
        }
    }
}